e2e-tests/production-runtime/package.json

```diff
@@ -42,9 +42,9 @@
     "start-server-and-test:locally": "start-server-and-test serve http://localhost:9000 serve-static-files http://localhost:8888 cy:open",
     "start-server-and-test:offline": "start-server-and-test serve http://localhost:9000 serve-static-files http://localhost:8888 cy:run:offline",
     "cy:open": "cypress open",
-    "cy:open:offline": "npm run cy:open -- --env TEST_PLUGIN_OFFLINE=y",
+    "cy:open:offline": "npm run cy:open --env TEST_PLUGIN_OFFLINE=y",
     "cy:run": "npm run cy:run:normal && npm run cy:run:bot",
-    "cy:run:offline": "npm run cy:run:normal -- --env TEST_PLUGIN_OFFLINE=y",
+    "cy:run:offline": "npm run cy:run:normal --env TEST_PLUGIN_OFFLINE=y",
```

Comment:
Running `cy:run:offline` as it is in master in circleci via SSH gives following output:

```
root@69033c290a63:/tmp/tmp.JnowHyJLtI/e2e-tests/production-runtime# npm run cy:run:offline

> gatsby-starter-default@1.0.0 cy:run:offline
> npm run cy:run:normal -- --env TEST_PLUGIN_OFFLINE=y


> gatsby-starter-default@1.0.0 cy:run:normal
> node ../../scripts/cypress-run-with-conditional-record-flag.js --browser chrome


libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[6564:1012/164002.972495:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[6564:1012/164002.975420:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Cypress could not verify that this server is running:

  > http://localhost:9000

We are verifying this server because it has been configured as your baseUrl.

Cypress automatically waits until your server is accessible before running tests.

We will try connecting to it 3 more times...
We will try connecting to it 2 more times...
We will try connecting to it 1 more time...

Cypress failed to verify that your server is running.

Please start this server and then run Cypress again.
node:child_process:858
    throw err;
    ^

Error: Command failed: /tmp/tmp.JnowHyJLtI/e2e-tests/production-runtime/node_modules/.bin/cypress run --browser chrome --env TEST_PLUGIN_OFFLINE=y
    at checkExecSyncError (node:child_process:820:11)
    at Object.execFileSync (node:child_process:855:15)
    at Object.<anonymous> (/tmp/tmp.JnowHyJLtI/scripts/cypress-run-with-conditional-record-flag.js:27:14)
    at Module._compile (node:internal/modules/cjs/loader:1120:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1174:10)
    at Module.load (node:internal/modules/cjs/loader:998:32)
    at Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 5957,
  stdout: null,
  stderr: null
}

Node.js v18.6.0
```

after dropping extra `--`, I was able to get it to start running:
```
root@69033c290a63:/tmp/tmp.JnowHyJLtI/e2e-tests/production-runtime# npm run cy:run:offline

> gatsby-starter-default@1.0.0 cy:run:offline
> npm run cy:run:normal --env TEST_PLUGIN_OFFLINE=y


> gatsby-starter-default@1.0.0 cy:run:normal
> node ../../scripts/cypress-run-with-conditional-record-flag.js --browser chrome

libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[7664:1012/164211.044348:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[7664:1012/164211.053106:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.7.0                                                                          │
  │ Browser:        Chrome 105 (headless)                                                          │
  │ Node Version:   v18.6.0 (/usr/local/bin/node)                                                  │
  │ Specs:          53 found (1-production.js, accessibility.js, assets.js, babelrc.js, client-onl │
  │                 y-paths.js, compilation-hash.js, env-vars.js, gatsby-plugin-image.js, gatsby-s │
  │                 cript-both-callbacks.js, gatsby-script-duplicate-scripts.js, gatsby-script-inl │
  │                 ine-scripts.js, gatsby-...)                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  1-production.js                                                                (1 of 53)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating


  Production build tests
    ✓ should render properly (317ms)
    ✓ should navigate back after a reload (394ms)
    ✓ should work when visiting a page with direct URL entry or an external link (250ms)
    ✓ should show 404 page when clicking a link to a non-existent page route (434ms)
^C
```

which I just aborted

this is likely a change that happend between whatever `npm` version was used on node 14 and version of `npm` used on node 18

Reply:
hmmm, this is probably not right - npm@8 docs still mention `--` ( https://docs.npmjs.com/cli/v8/commands/npm-run-script ) so this probably just now doesn't do anything instead of passing it to underlying scripts - this need further debugging

Reply:
it does unblock at least running something - so I do think `--` is at least part of the problem that is happening, but the change might not be the right one